File: original artifact/Ian_Foss_cs-360_project_three_inventory_app/cs360_project_three/app/src/main/java/com/example/cs_360_project_three/UserEntity.java
package com.example.cs_360_project_three;

import androidx.room.Entity;
import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;

import androidx.room.PrimaryKey;


@Entity(tableName = "user_table")
public class UserEntity {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    private long id;

    @NonNull
    @ColumnInfo(name = "username")
    private String username;

    @NonNull
    @ColumnInfo(name = "password")
    private String password;



    public long getId() {return id;}
    public @NonNull String getUsername() {return username;}
    public @NonNull String getPassword() {return password;}


    public void setId(long id) { this.id = id; }
    public void setUsername(@NonNull String username) {this.username = username;}
    public void setPassword(@NonNull String password) {this.password = password;}


}

File: enhancements/Enhancement One/CS-499_WIP/app/src/main/java/com/example/CS499_Capstone/UserEntity.kt
package com.example.CS499_Capstone

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "user_table",
    indices = [Index(value = ["username"], unique = true)]
)
data class UserEntity (
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    val id: Long = 0,

    @JvmField
    @ColumnInfo(name = "username")
    val username: String,

    @JvmField
    @ColumnInfo(name = "password")
    val password: String
)
